src/lib.rs

```diff
@@ -1,185 +1,30 @@
-//! ```
-//! If you're still using Rust 2015, add this instead:
-//! ```rust,edition2015
-//! extern crate core;
-//! #[macro_use]
-//! extern crate derive_more;
-//! # fn main() {} // omit wrapping statements above into `main()`
-//! ```
-//!
-//! [`cargo-expand`]: https://github.com/dtolnay/cargo-expand
-//! [`derive-new`]: https://github.com/nrc/derive-new
-//!
-//! [`From`]: https://jeltef.github.io/derive_more/derive_more/from.html
-//! [`Into`]: https://jeltef.github.io/derive_more/derive_more/into.html
-//! [`FromStr`]: https://jeltef.github.io/derive_more/derive_more/from_str.html
-//! [`TryInto`]: https://jeltef.github.io/derive_more/derive_more/try_into.html
-//! [`IntoIterator`]: https://jeltef.github.io/derive_more/derive_more/into_iterator.html
-//! [`AsRef`]: https://jeltef.github.io/derive_more/derive_more/as_ref.html
-//! [`AsMut`]: https://jeltef.github.io/derive_more/derive_more/as_mut.html
-//!
-//! [`Display`-like]: https://jeltef.github.io/derive_more/derive_more/display.html
-//!
-//! [`Error`]: https://jeltef.github.io/derive_more/derive_more/error.html
-//!
-//! [`Index`]: https://jeltef.github.io/derive_more/derive_more/index_op.html
-//! [`Deref`]: https://jeltef.github.io/derive_more/derive_more/deref.html
-//! [`Not`-like]: https://jeltef.github.io/derive_more/derive_more/not.html
-//! [`Add`-like]: https://jeltef.github.io/derive_more/derive_more/add.html
-//! [`Mul`-like]: https://jeltef.github.io/derive_more/derive_more/mul.html
-//! [`Sum`-like]: https://jeltef.github.io/derive_more/derive_more/sum.html
-//! [`IndexMut`]: https://jeltef.github.io/derive_more/derive_more/index_mut.html
-//! [`DerefMut`]: https://jeltef.github.io/derive_more/derive_more/deref_mut.html
-//! [`AddAssign`-like]: https://jeltef.github.io/derive_more/derive_more/add_assign.html
-//! [`MulAssign`-like]: https://jeltef.github.io/derive_more/derive_more/mul_assign.html
-//!
-//! [`Constructor`]: https://jeltef.github.io/derive_more/derive_more/constructor.html
-//! [`IsVariant`]: https://jeltef.github.io/derive_more/derive_more/is_variant.html
-//! [`Unwrap`]: https://jeltef.github.io/derive_more/derive_more/unwrap.html
-
+//! [`From`]: crate::From
+//! [`Into`]: crate::Into
+//! [`FromStr`]: crate::FromStr
+//! [`TryInto`]: crate::TryInto
+//! [`IntoIterator`]: crate::IntoIterator
+//! [`AsRef`]: crate::AsRef
+//!
+//! [`Display`-like]: crate::Display
+//!
+//! [`Error`]: crate::Error
+//!
+//! [`Index`]: crate::Index
+//! [`Deref`]: crate::Deref
+//! [`Not`-like]: crate::Not
+//! [`Add`-like]: crate::Add
+//! [`Mul`-like]: crate::Mul
+//! [`Sum`-like]: crate::Sum
+//! [`IndexMut`]: crate::IndexMut
+//! [`DerefMut`]: crate::DerefMut
+//! [`AddAssign`-like]: crate::AddAssign
+//! [`MulAssign`-like]: crate::MulAssign
+//!
+//! [`Constructor`]: crate::Constructor
+//! [`IsVariant`]: crate::IsVariant
+//! [`Unwrap`]: crate::Unwrap
+#![doc = include_str!("../README.md")]
+#![cfg_attr(docsrs, feature(doc_auto_cfg))]
```

Comment:
What is this suppossed to do?

Reply:
Okay after a bit of research and playing around, I realised it's to shows the `feature` that enables a derive. Is this the only thing, or am I missing something. (showing the feature does seem very useful btw).

Reply:
@JelteF yup, this is solely for the "Available on crate feature `<feature_name>` only" badges on `docs.rs`.